### PR TITLE
Autogenerate optionSets and f01MphssBaseline

### PR DESCRIPTION
**Pull request by mtuchi**
**Description**
Fetch metadata from sharepoint then generate f01MphssBaseline and optionSets

**Fixes**
- #47

### CODE REVIEW

File: workflows/wf3/2-map-metadata.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+  return state;
+});
+
+fn(state => {
+  const { optionSetsXlsData } = state;
+  const keys = optionSetsXlsData[1];
+
+  const optsMap = optionSetsXlsData.slice(2).map(item => {
```
Comment (josephjclark) [inline workflows/wf3/2-map-metadata.js:42]: Is this function the same for each block? If so you should declare it as a const at the top of the file. It'll be more readable too

File: workflows/wf3/2-map-metadata.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    const obj = item.reduce((acc, value, idx) => {
+      acc[keys[idx]] = value;
+      return acc;
+    }, {});
+
+    return obj;
+  });
+
+  state.optionSets = optsMap
+    .filter(
```
Comment (josephjclark) [inline workflows/wf3/2-map-metadata.js:52]: I think this filter is common too? So I would hoist this out as well

**Review by josephjclark — COMMENTED** [review]
Just a couple of small suggestions

File: workflows/wf3/2-map-metadata.js (hunk 2 above)
Comment (mtuchi) [reply workflows/wf3/2-map-metadata.js:52]: The filter is different for `optionSets` we want to make sure `External ID` and `DHIS2 DE UID` while for `f01MhpssBaseline` it's `External ID` and `DHIS2 DE full name`

**Review by josephjclark — APPROVED** [review]

**Pull request merged**